rust_dev_preview/lambda/src/bin/list-all-function-runtimes.rs

```diff
@@ -3,60 +3,66 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-use aws_config::meta::region::RegionProviderChain;
-use aws_sdk_lambda::{Client, Error, Region, PKG_VERSION};
+use aws_sdk_lambda::Error;
+use lambda_code_examples::{make_client, Opt as BaseOpt};
 use structopt::StructOpt;
 
 #[derive(Debug, StructOpt)]
 struct Opt {
-    /// The AWS Region in which the client is created.
-    #[structopt(short, long)]
-    region: Option<String>,
+    #[structopt(flatten)]
+    base: BaseOpt,
 
     /// Just show runtimes for indicated language.
     /// dotnet, go, node, java, etc.
     #[structopt(short, long)]
     language: Option<String>,
-
-    /// Whether to display additional runtime information.
-    #[structopt(short, long)]
-    verbose: bool,
 }
 
 /// Lists the ARNs and runtimes of all Lambda functions in all Regions.
 // snippet-start:[lambda.rust.list-all-function-runtimes]
-async fn show_lambdas(verbose: bool, language: &str, reg: &'static str) {
-    let region_provider = RegionProviderChain::default_provider().or_else(reg);
-    let config = aws_config::from_env().region(region_provider).load().await;
-    let client = Client::new(&config);
-
-    let resp = client.list_functions().send().await;
-    let functions = resp.unwrap().functions.unwrap_or_default();
-    let max_functions = functions.len();
-    let mut num_functions = 0;
-
-    for function in functions {
-        let rt_str: String = String::from(function.runtime().unwrap().as_ref());
-        // If language is set (!= ""), show only those with that runtime.
-        let ok = rt_str
-            .to_ascii_lowercase()
-            .contains(&language.to_ascii_lowercase());
-        if ok || language.is_empty() {
-            println!("  ARN:     {}", function.function_arn().unwrap());
-            println!("  Runtime: {}", rt_str);
-            println!();
-
-            num_functions += 1;
-        }
+async fn show_lambdas(language: &str, region: &str, verbose: bool) -> Result<(), Error> {
+    let (client, _) = make_client(BaseOpt {
```

Comment:
Resolving the `SdkConfig` is something that should only be done once, ideally. It does a considerable amount of work to resolve configuration from files, env vars, IMDS, etc.

Reply:
For this example, we're iterating regions to find all functions across the account. While this isn't the common case, it's certainly a valuable case. Is there a way we could make it easier and cheaper to reopen a Config to modify it?

```
SdkConfig::Builder::reopen(sdk_config).region(new_region).build();
```

Reply:
It seems like we could add `SdkConfig::to_builder(&self)` that allows a new config to be built with some changes made.

Reply:
Much better name :)